File: 0011-container-with-most-water/0011-container-with-most-water.ts
function maxArea(height: number[]): number {
  let result = 0;
  let left = 0;
  let right = height.length - 1;
  
  
  while (left <= right) {
    const newArea = (right - left) * Math.min(height[left], height[right]);
    result = Math.max(result, newArea);
    
    if (height[left] > height[right]) {
      right--;
    } else {
      left++;
    }
  }
  
  return result;
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        result = 0

        while left < right:
            current_width = right - left
            current_height = min(height[left], height[right])
            result = max(result, current_width * current_height)
            
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return result
